src/function_benchmark.rs

```diff
@@ -0,0 +1,145 @@
+use colored::Colorize;
+use std::{fmt, time::Duration};
+
+const RUNTIME_THRESHOLD: Duration = Duration::from_millis(5);
+
+pub struct FunctionBenchmark {
+    pub runtime: Duration,
+}
+
+impl FunctionBenchmark {
+    pub fn new(runtime: Duration) -> Self {
+        FunctionBenchmark { runtime }
+    }
+}
+
+impl fmt::Display for FunctionBenchmark {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        let title = "      Benchmark Results      ".black().on_bright_green();
+        write!(f, "{}\n\n", title)?;
+
+        let runtime_display: String = if self.runtime <= RUNTIME_THRESHOLD {
+            format!("{:?}", self.runtime).bright_green().to_string()
+        } else {
+            format!(
+                "{:?} <- maximum allowed is {:?}",
+                self.runtime, RUNTIME_THRESHOLD
+            )
+            .red()
+            .to_string()
+        };
+
+        writeln!(f, "Runtime: {}", runtime_display)?;
+
+        Ok(())
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use anyhow::anyhow;
+    use std::{
+        path::{Path, PathBuf},
+        time::Instant,
+    };
+    use wasmtime::*;
+    use wasmtime_wasi::WasiCtxBuilder;
+
+    #[test]
+    fn test_benchmark_runtime_allowed() {
```

Comment:
Hmm I'm not sure the value of these tests. Could you explain your thought for why we need these tests? 

There isn't really any business logic to test. And these tests don't cover the usage in the real case. We are also duplicating all the script running functionality in here which is another place to maintain.

Reply:
`FunctionBenchmark` isn't doing any timing though. That happens with `Instant::now` and `now.elapsed`. `FunctionBenchmark` just takes in the data and does some printing, which is why I don't think the tests are covering anything that this struct is doing. I don't think you need to have tests for how things get printed, that can be verified visually. I think what's here is simple enough for no tests